barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/ultra_circuit_builder.cpp

```diff
@@ -716,44 +729,40 @@ std::vector<uint32_t> UltraCircuitBuilder_<ExecutionTrace>::decompose_into_defau
                 -1,
                 0,
             },
-            ((i == num_limb_triples - 1) ? false : true));
-        // TODO(https://github.com/AztecProtocol/barretenberg/issues/1450): this is probably creating an unused
-        // wire/variable in the circuit, in the last iteration of the loop.
-        accumulator_idx = this->add_variable(fr(new_accumulator));
-        accumulator = new_accumulator;
+            (i != num_limb_triples - 1));
+        if (i != num_limb_triples - 1) {
+            accumulator_idx = this->add_variable(fr(new_accumulator));
+            accumulator = new_accumulator;
+        }
     }
     return sublimb_indices;
 }
 
-/**
- * @brief Constrain a variable to a range
- *
- * @details Checks if the range [0, target_range] already exists. If it doesn't, then creates a new range. Then tags
- * variable as belonging to this set.
- *
- * @param variable_index
- * @param target_range
- */
 template <typename ExecutionTrace>
-void UltraCircuitBuilder_<ExecutionTrace>::create_new_range_constraint(const uint32_t variable_index,
-                                                                       const uint64_t target_range,
-                                                                       std::string const msg)
+void UltraCircuitBuilder_<ExecutionTrace>::create_small_range_constraint(const uint32_t variable_index,
+                                                                         const uint64_t target_range,
+                                                                         std::string const msg)
 {
+    // make sure `target_range` is not too big.
+    BB_ASSERT_GTE(MAX_SMALL_RANGE_CONSTRAINT_VAL, target_range);
     const bool is_out_of_range = (uint256_t(this->get_variable(variable_index)).data[0] > target_range);
     if (is_out_of_range && !this->failed()) {
         this->failure(msg);
     }
     if (range_lists.count(target_range) == 0) {
         range_lists.insert({ target_range, create_range_list(target_range) });
     }
-
+    // The tag of `variable_index` is `DUMMY_TAG` if it has never been range-constrained and a non-trivial value
+    // otherwise.
     const auto existing_tag = this->real_variable_tags[this->real_variable_index[variable_index]];
     auto& list = range_lists[target_range];
 
-    // If the variable's tag matches the target range list's tag, do nothing.
+    // If the variable's tag matches the target range list's tag, do nothing; the variable has _already_ been
+    // constrained to this exact range (i.e., `create_new_range_constraint(variable_index, target_range)` has already
+    // been called).
     if (existing_tag != list.range_tag) {
```

Comment:
This is just a style thing but in situations like this I find it nicer to do 
```
if (existing_tag == list.range_tag) {
    return;
}    
```
(which has a lovely self-documenty feel to it) then the remaining code can be pulled back 1 level. Then again, this is an audit and such opinionated style changes may not be worth it :)

Reply:
Then again *again*, this is a 4-deep conditional which is pretty ick